### PR TITLE
wip: Trying to make chumsky 1.0.0 work

### PR DESCRIPTION
**Pull request by sirasistant**
# Description

## Problem\*

Resolves 

## Documentation\*

Check one:
- [ ] No documentation needed.
- [ ] Documentation included in this PR.
- [ ] **[Exceptional Case]** Documentation to be submitted in a separate PR.

# PR Checklist\*

- [ ] I have tested the changes locally.
- [ ] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

File: compiler/noirc_errors/src/position.rs [hunk 1]
```diff
@@ -109,12 +109,17 @@ impl From<Range<u32>> for Span {
...
 
     type Offset = u32;
 
-    fn new(_context: Self::Context, range: Range<Self::Offset>) -> Self {
+    fn new(_context: Self::Context, mut range: Range<Self::Offset>) -> Self {
+        // TODO why chumsky reports reversed ranges??
+        if range.start > range.end {
+            range = Range { start: range.end, end: range.start };
+        } else {
+        }
```
Comment (jfecher) [inline compiler/noirc_errors/src/position.rs:122]: This has been a problem in the past when using certain error recovery strategies I think. Hopefully it hasn't gotten more common with 1.0 😅

**Review by jfecher — COMMENTED** [review]
Since this is not the first effort to migrate to 1.0, I think it's alright if we have to remove a few of our existing parser recovery tests to make this work. We'll re-evaluate / rework recovery at a later date anyway. IIRC last time we tried to migrate to 1.0 we got everything working sans 2 parser recovery tests.

**Comment by michaeljklein** [comment]
Because release candidates for `chumsky 1.0.0` are failing to build and/or untagged on github, I'm holding off on continuing on this for now.
- The `.0` release candidate of `chumsky 1.0.0` is the latest tagged version on github, but it fails to compile
- The latest version with a tag is pre-`1.0.0` (`0.9.3`)

**Comment by TomAFrench** [comment]
Closing as this branch seems abandoned

**Pull request closed without merging**